src/sync.cpp

```diff
@@ -3040,6 +3058,43 @@ error JSONSyncConfigIOContext::read(const LocalPath& dbPath,
     return API_OK;
 }
 
+error JSONSyncConfigIOContext::remove(const LocalPath& dbPath,
+                                      const unsigned int slot)
+{
+    using std::to_string;
+
+    LocalPath path = dbPath;
+
+    // Generate the rest of the path.
+    path.appendWithSeparator(mName, false);
+    path.append(LocalPath::fromPath("." + to_string(slot), mFsAccess));
+
+    return mFsAccess.unlinklocal(path) ? API_OK : API_EWRITE;
```

Comment:
Perhaps we should post some log message in case of failure. The callers don't post any message. Actually, the upper caller (`MegaClient::removeCaches()` doesn't check if there's been an error or not.

Reply:
This function now logs when it is unable to remove the specified slot.